src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ClassDataContract.cs

```diff
@@ -439,16 +439,12 @@ static internal bool IsNonAttributedTypeValidForSerialization(Type type)
             }
         }
 
-        private static string[] s_knownSerializableTypeNames = new string[] {
-                "System.Collections.Queue",
-                "System.Collections.Stack",
-                "System.Globalization.CultureInfo",
-                "System.Version",
+        private static string[] s_knownSerializableGenericTypeNames = new string[] {
```

Comment:
Why is parsing this and converting it to a Dictionary even necessary? Why not a pre-parsed dictionary? e.g.:

```csharp
private static readonly Dictionary<string, string[]> s_knownSerializableGenericTypes = new Dictionary<string, string[]>
{
    { "System.Collections.Generic.KeyValuePair`2", Array.Empty<string>() },
    { "System.Collections.Generic.Queue`1", new[] { "_syncRoot" } },
    { "System.Collections.Generic.Stack`1", new[] { "_syncRoot" } },
    { "System.Collections.ObjectModel.ReadOnlyCollection`1", new[] { "_syncRoot" } },
    { "System.Collections.ObjectModel.ReadOnlyDictionary`2", new[] { "_syncRoot", "_keys", "_values" } },
    { "System.Tuple`1", Array.Empty<string>() },
    
    ... etc. ...
};
```